### DISCUSSION TITLE
Tanstack Table  -- serverside

### DISCUSSION THREAD

**Discussion by Er4twXz** [TanStack/table#5410] [Ideas] [answered]
As far as I understand Tanstack table currently can only be used in nextjs client components because of the hooks and not in server side components.

In order to be SEO friendly I would prefer to use serverside components.

Any chance to see this in future ? If this is a stupid idea please feel fry to close this issue.

**Comment by KevinVandy** [comment, marked as answer]
"use client" !== client only.

I suggest understanding how next.js works. Client components still run on the server in next.js. SEO still there. The only way to mess up SEO/SSR in next.js is to use dynamic lazy imports for components.

**Reply by Er4twXz** [reply]
Thanks for your reply. As I said -- ? If this is a stupid idea please feel free to close this issue.

"use client" was my missunderstanding

https://nextjs.org/docs/app/building-your-application/rendering/client-components
